components/StateHandlers/InitialStateHandler.tsx

```diff
@@ -25,84 +26,106 @@ const getSchemedURLFromUniversalURL = (url: string) => {
   return schemedURL;
 };
 
-export let initialURL = "";
+const isDevelopmentClientURL = (url: string) => {
+  return url.includes("expo-development-client");
+};
 
 export default function InitialStateHandler() {
   const colorScheme = useColorScheme();
-  const setDesktopConnectSessionId = useOnboardingStore(
-    (s) => s.setDesktopConnectSessionId
-  );
-  const { setSplashScreenHidden, hydrationDone } = useAppStore(
-    useSelect(["setSplashScreenHidden", "hydrationDone"])
+
+  const url = Linking.useURL();
```

Comment:
using useURL that way I'm not sure you will handle all cases because it will "remember" the last deeplink forever
Example : 
- be logged into the app on (only) one account
- click on a dm deeplink / universal link => opens converse on the page of the convo
- logout from the account without killing the app
- log back it to the account (or another account with profile already setup)
- you will be redirected to the convo while you clearly didn't just click on the link! That's because the system remembers that deeplink

It's not a very big issue for DMs (but still annoying) but could be for other stuff (I remember desktop login had the issue)

Reply:
Normally this should be okay because the url won't change so won't trigger the useEffect below.

BUT you're right that we're also listening to check if the user is signed in so this can make the useEffect retrigger. Will push a fix

Reply:
@thierryskoda the flow I described is one I experienced! useEffect below will be triggered the `isSignedIn` will go to true then false then back to true because you logout and relogin.